plugin/forward/proxy.go

```diff
@@ -63,6 +63,9 @@ func (p *Proxy) SetTLSConfig(cfg *tls.Config) {
 	p.client = dnsClient(cfg)
 }
 
+// TLSConfig provides current TLS config.
+func (p *Proxy) TLSConfig() *tls.Config { return p.transport.tlsConfig }
+
```

Comment:
is it safer to return a copy  instead of a pointer ? 
I mean : can someone modify the tls.Config using this function ?

Reply:
Yes. As it is pointer and it is possible to modify it.
As it may be nil, I will return copy with bool "ok" variable.